### PR TITLE
Updated for 11.0 The War Within

### PR DESCRIPTION
**Pull request by bpapuni**
With The War Within Blizzard have made some changes to how macros work. 

The length of macros executable by "macrotext" is being restricted to 255 characters, and all macros (both real and "macrotext" based ones) are being restricted to remove the ability to chain macros. This means that one macro cannot "/click" a button that itself would execute another macro.
https://github.com/Stanzilla/WoWUIBugs/issues/552#issuecomment-2192109095

This pull request changes several constants to handle this change. Additionally, it moves refactors several functions into the appropriate namespaces Blizzard has introduced.

### CODE REVIEW

**Comment by Sellorio** [comment]
Is there nothing that can be done about the 255 character limit? So ratty of blizzard to do this.

File: src/engine/mega-macro-action-bar-engine.lua [hunk 1]
```diff
@@ -397,7 +392,7 @@ end
 function MegaMacroActionBarEngine.OnUpdate(elapsed)
     UpdateRangeTimer(elapsed)
 
-    local focus = GetMouseFocus()
+    local focus = GetMouseFoci()[1]
```
Comment (Sellorio) [inline src/engine/mega-macro-action-bar-engine.lua:395]: Hmm are they setting the stage for mobile with this or is it an object layering thing?

**Comment by Sellorio** [comment]
When users edit a macro over 255 characters (i.e. they update to this version and they have an old macro that was larger), will it still show the full macro and let them edit it down to size or will it just clip it off?

**Comment by bpapuni** [comment]
> Is there nothing that can be done about the 255 character limit? So ratty of blizzard to do this.

Unfortunately not.

> When users edit a macro over 255 characters (i.e. they update to this version and they have an old macro that was larger), will it still show the full macro and let them edit it down to size or will it just clip it off?

I actually hadn't considered this. In it's current state it'll just clip it off. I'll rework the behavior tonight when I get home from work to show the full macro and allow the user to edit it down.

File: src/engine/mega-macro-action-bar-engine.lua (hunk 1 above)
Comment (bpapuni) [reply src/engine/mega-macro-action-bar-engine.lua:395]: I believe just an object layering thing

**Pull request merged**